src/libraries/System.Net.WebSockets.Client/tests/ConnectTest.Http2.cs

```diff
@@ -227,5 +227,47 @@ await Http2LoopbackServer.CreateClientAndServerAsync(async uri =>
                 await connection.SendResponseHeadersAsync(streamId2, endStream: false, HttpStatusCode.OK);
             }, new Http2Options() { WebSocketEndpoint = true, UseSsl = false });
         }
+
+        [OuterLoop("Uses external servers", typeof(PlatformDetection), nameof(PlatformDetection.LocalEchoServerIsNotAvailable))]
+        [Theory]
+        [MemberData(nameof(EchoServers))]
+        [SkipOnPlatform(TestPlatforms.Browser, "System.Net.Sockets is not supported on this platform")]
+        public async Task ConnectAsync_Http11WithRequestVersionOrHigher_DowngradeSuccess(Uri server)
+        {
+            using (var cws = new ClientWebSocket())
+            using (var cts = new CancellationTokenSource(TimeOutMilliseconds))
+            {
+                cws.Options.HttpVersion = HttpVersion.Version11;
+                cws.Options.HttpVersionPolicy = HttpVersionPolicy.RequestVersionOrHigher;
+                await cws.ConnectAsync(server, GetInvoker(), cts.Token);
+                Assert.Equal(WebSocketState.Open, cws.State);
+            }
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.Browser, "System.Net.Sockets is not supported on this platform")]
+        public async Task ConnectAsync_Http11WithRequestVersionOrHigher_Loopback_Success()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (var cws = new ClientWebSocket())
+                using (var cts = new CancellationTokenSource(TimeOutMilliseconds))
+                {
+                    cws.Options.HttpVersion = HttpVersion.Version11;
+                    cws.Options.HttpVersionPolicy = HttpVersionPolicy.RequestVersionOrHigher;
+
+                    Uri wsUrl = new Uri(url.ToString().Replace("http://", "ws://"));
```

Comment:
you need to pass `LoopbackServer.Options` with `WebSocketEndpoint = true`, then you won't need to modify the url

also, you should ensure you use ssl in this test, or HTTP/2 would not even be attempted, so the test won't validate the issue. you can do this also in `LoopbackServer.Options`, with `UseSsl = true`